modules/gitops-additional-permissions-for-cluster-config.adoc

```diff
@@ -0,0 +1,69 @@
+// Module included in the following assembly:
+//
+// * configuring-an-openshift-cluster-by-deploying-an-application-with-cluster-configurations.adoc
+
+:_content-type: PROCEDURE
+[id="gitops-additional-permissions-for-cluster-config_{context}"]
+= Providing additional permissions for cluster configuration
+
+For cluster-configuration management you can add permissions to an Argo CD instance. 
+
+.Procedure
+
+Create **Cluster Roles** with additional permissions and then a new **Cluster Role Binding** to associate them to a **ServiceAccount**:
+
+. Log in to the OpenShift Container Platform web console as an admin.
+. In Openshift console, go to **User Management** -> **Roles** -> **Create Role**. Use the following `ClusterRole` yaml template to add rules to specify the additional permissions.
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: secrets-cluster-role
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["*"]
+----
++
+. Click **Create**. After adding the cluster role, set the **ClusterRoleBinding**.
+. In Openshift console, go to **User Management** -> **Role Bindings** -> **Create Binding**.
+. Select **All Projects** from the **Project** drop-down.
+. Click **Create binding**.
+. Select **Binding type** as **Cluster-wide role binding (ClusterRoleBinding)**.
+. Enter the **RoleBinding name**, the name should be unique.
+. Select the new cluster role created or any existing cluster role from the drop down list.
+. Select the **Subject** as **ServiceAccount** and the provide the **Subject Namespace** and **Name**.
+.. **Subject Namespace**: openshift-gitops
+.. **Subject Name**: openshift-gitops-argocd-application-controller
+. Click **Create**. The yaml file after adding the role binding is as follows:app-name:
++
+[source,yaml]
+----
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cluster-role-binding
+  uid: 841ecedb-f059-41f1-b6f9-c19e3e0d53c2
+  resourceVersion: '79371'
+  creationTimestamp: '2022-07-12T05:51:44Z'
+  managedFields:
+    - manager: Mozilla
+      operation: Update
+      apiVersion: rbac.authorization.k8s.io/v1
+      time: '2022-07-12T05:51:44Z'
+      fieldsType: FieldsV1
+      fieldsV1:
+        'f:roleRef': {}
+        'f:subjects': {}
```

Comment:
These fields should not be part of the source resource (e.g. should not be used to create the resource).